backend/logic/users/users.js

```diff
@@ -277,12 +244,126 @@ var login = async (username, password) => {
   }
 };
 
+var assignStudentMajor = async (studentID, major) => {
+  let error = false;
+
+  if (!format.verifyStudentId(studentID)) {
+    error = "Invalid student id";
+  } else if (!format.verifyCurriculumName(major)) {
+    error = "Invalid curriculum name";
+  }
+
+  if (!error == false) {
+    console.error(error);
+    throw new Error(error);
+  }
+
+  let ifUserExist;
+  let ifMajorExist;
+  let existingMajor;
+
+  let conn = await mysql.getNewConnection();
+  try {
+    ifUserExist = await conn.query(`SELECT COUNT(*) AS count FROM students WHERE student_id = ?;`, [studentID]);
+    ifMajorExist = await conn.query(`SELECT COUNT(*) AS count FROM curriculums WHERE curriculum_name = ?;`, [major]);
+    existingMajor = await conn.query(`SELECT COUNT(*) AS count FROM student_majors WHERE student_id = ?;`, [studentID]);
+  } catch (err) {
+    console.log(err);
```

Comment:
Minor suggestion, since you are creating a new connection later, you can put `conn.release()` into `catch` block, so that you don't have to do ```conn = await mysql.getNewConnection();``` again